modules/fedimint-mint/src/lib.rs

```diff
@@ -441,6 +441,18 @@ impl ServerModule for Mint {
         MintDecoder
     }
 
+    fn database_version(&self) -> DatabaseVersion {
+        DatabaseVersion { version: 1 }
+    }
+
+    async fn migrate_database(
+        &self,
+        _db: &Database,
+        _db_version: u64,
```

Comment:
I'm surprised this parameter is needed at all. Shouldn't each module source code know what its latest database schema version is, and be able to read the current schema from the database?


I kind of think that we should just let modules call `migrate_database` automatically when instantiatied. But maybe there are some reasons not to. But we can add it easily when we need it, I guess.

Reply:
The module source code will know the current database version (hardcoded in the code) and it will also know the database version that it read from disk (that's what this is, it needs a better name). Perhaps this should just be read from disk here instead of being passed as a parameter.

The module will need to use the database version from disk to know how to read the structs from disk because they might be in a different format from the current code.

I'm indifferent if we should explicitly call `migrate_database` or just let the modules do it. I suppose by adding it to the ServerModule trait, it would force module developers to think about database migrations.

Reply:
Removed migrate_database from the trait and calling it from init instead